### ISSUE TITLE
Dashboards: Dashboard changed dialog appears in viewer mode

### ISSUE THREAD

**Issue by rpetersen27** [grafana/grafana#71119]
### 
# What went wrong?

**What happened**:

- We have customers with monitors, where we have a preconfigured URL to show some dashboards. 
- The url looks as follows: `?orgId=1&refresh=1m&var-MaxJobs=2&var-JobId=All&var-Tracks=All&var-Line=L07`
- The anonymous authentication is enabled and default role is set to "viewer"
- When another session with Editor rights modifies the dashboard, the viewer will receive the "Dashboard changed" dialog. Since this is displayed on monitors without keyboard/mouse, these dashboards are stuck in that screen until the browser is manually refreshed. 
- As you can see, we set the refresh=1m. The "Dashboard changed" dialog will only appear after the first automatic refresh has occurred. If you manage to change the dashboard before the automatic refresh occurred, the dialog will not appear and the site will be refreshed correctly. 

**What did you expect to happen**:

  - No "Dashboard changed"-dialog, just a page refresh. 

### 
# How do we reproduce it?

**Step 1**:

- Open Grafana as editor in one session and another session with anonymous auth as viewer with an automatic refresh interval. 

**Step 2**:

- Wait until the viewer has refreshed automatically.

**Step 3**:

- As the editor, edit and save the dashboard. 

**Step 4**:

- Switch back to the viewer session and observe the "Dashboard changed" dialog

### 
# What Grafana version are you using?

v9.4.3, v10.0.1

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

Docker

### User's OS?

Balena OS

### User's Browser?

Google Chrome 114.0.5735.198 (official build), but happens in all tested browsers

### Is this a Regression?

None

### Anything else to add?

We experienced this issue at first after upgrading to 9.4.1 (from 8.something). But I cannot assure that the issue hasn't been there before the migration. So I am not sure if this is a regression.

**Comment by evictorero** [comment]
Hi @rpetersen27 , thanks for reporting this.
I couldn't reproduce it. The modal with the title **Dashboard changed** appears only when the viewer changes the dashboard in order to avoid losing local changes (which in this case does not make any sense since it is a viewer). 
Can you tell me more about which panels are you using and which data sources? 
I could reproduce it by only selecting one specific time-series in the anonymous viewer, which "changes" the dashboard but from a vanilla dashboard I always receive the alert "Dashboard updated" instead of the modal (which is expected).
This modal shouldn't appear unless you modify the viewer dashboard
<img width="594" alt="image" src="https://github.com/grafana/grafana/assets/20256983/44ce1235-5665-4d06-a854-97d2a089237c">

This alert is correct
<img width="425" alt="image" src="https://github.com/grafana/grafana/assets/20256983/f5cae3d6-019f-49d4-9679-40c774e5775f">

**Comment by rpetersen27** [comment]
Hi @evictorero, thank you for your response. 

I tried to reproduce this with a new dashboard and I was not able to reproduce it. I have one specific dashboard, where I can reproduce it consistently after the auto refresh has run for the viewer (nothing else was changed). It is important to note, that this is a quite old dashboard and had been migrated over several versions. 
It is enough to just add a new text-panel or move an existing one. I tried to create a minimal dashboard based on this broken dashboard, that has this error. 

I've been running this with the following setup:
* Grafana 9.4.1 in docker
* Following environment variables are set:
  * GF_AUTH_ANONYMOUS_ENABLED=true
  * GF_AUTH_ANONYMOUS_ORG_ROLE=Viewer
  * GF_AUTH_ANONYMOUS_ORG_NAME=Heinz-Glas
* I am using a postgres-db as datasource. I configured all variables in a way that they don't query real data on the DB but mimic the behaviour from the original dashboard. So you just have to setup any postgres datasource.

Steps to reproduce  
* I open the dashboard in chrome browser. One tab, I log in so I am able to edit the dashboard. I don't have any params set, so the dashboard will be shown for Line L06
* Second tab is in incognito-mode in chrome, with url params: `?orgId=1&refresh=5s&var-Line=L07`
* I wait for the second dashboard to do the auto refresh (5 seconds)
* I go to the editable dashboard and just grab the html with "Track" and drag it somewhere else. 
* I save the dashboard and go back to the viewer dashboard in the incoqnito-tab
* The dashboard shows the "Dashboard changed"-Dialog

Here is the dashboard: 
[Broken Dashboard-1689074793288.json.txt](https://github.com/grafana/grafana/files/12014217/Broken.Dashboard-1689074793288.json.txt)

I hope this is reproducible for you as well now. If you need any further information, please let me know.

**Comment by axelavargas** [comment]
Hey @rpetersen27 👋🏾, just a friendly reminder - please let us know if the issue is still present 😄.

**Comment by evictorero** [comment]
Is it possible that the URL is opening with a non-default value? Is `L07` the default value of var `Line`? I could reproduce it in Grafana 10 but only happen when the value informed in the URL is not the default one.
This is the docker-compose I'm using 
[docker-compose.yml.txt](https://github.com/grafana/grafana/files/12362025/docker-compose.yml.txt)

**Comment by rpetersen27** [comment]
Yes, that seems to be the case. When I open the dashboard without any URL-params I get `L06`, so I assume that is the default value. In my case, I used `L07` in the "read-only" tab, which differs from the default one. 

The docker setup looks similar to mine. 

Glad, that you can reproduce it. If you need any more information, just let me know.
